src/com/geekhub/hw3/auth/AuthenticationService.java

```diff
@@ -24,7 +24,21 @@ public static void main(String[] args) {
     };
 
     private User auth(String login, String password) throws AuthException {
-        //TODO: Implement me
+        if (login == null || password==null || login.isEmpty() || password.isEmpty()) throw new WrongCredentialsException("wrong credentials");
+        User user = getUserByLogin(login);
+        if (user != null) {
+            if (user.getPassword().equals(password)) {
```

Comment:
You can also check whether user, you get from repository, has password, before call equals method.